doc/aaa/TACACS+ Server Monitor.md

```diff
@@ -0,0 +1,80 @@
+# TACACS+ server monitor design
+
+## Overview
+
+SONiC device usually configured with multiple TACACS+ server, when a server is unreachable, SONiC device will try to connect with next TACACS+ server.
+
+SONiC device will communicate with TACACS+ server in following scenarios:
+1. Remote user login to SONiC device.
+2. Remote user run commands on SONiC device.
+
+There is a timeout for each server, the default value is 5 seconds, this means if the first server not reachable, SONiC device will stuck there when user login or running commands.
+
+To improve this issue, SONiC will add a TACACS+ server monitor to change server priority, a server unreachable or slow response will be downgrade.
+
+### Functional Requirement
+- Monit TACACS+ server unreachable event from syslog.
```

Comment:
>unreachable event from syslog [](http://example.com/codeflow?start=23&length=29)

Monitor local syslog file is not a robust method. How about improve pam component with more counters (possibly inside CountersDB)?

Reply:
Fixed, will test TACACS server connection periodically and update latency data to COUNTER_DB